### ISSUE TITLE
code - app_cover component

### ISSUE THREAD

**Issue by Mehrabbruno** [dat-ecosystem/dat-ecosystem#6]
#3 

## `todo`

- [x] code - app_cover_v0.0.1 component
   - `@input` 📦 [`home_page_v0.0.10`](https://www.figma.com/file/1zBxgchAmrDGw2FDzu7uxX/dat-ecosystem-website?type=design&node-id=1760-2784&t=i9jWdxjECWAyb6ri-4) from [dat-ecosystem website_homepage](https://github.com/dat-ecosystem/organization/issues/85)
   - [x] code the app_cover component
   - [x] update app_cover by connecting it to window_bar using notify
   - `@output` 📦 [`app_cover_v0.0.1`]() from [comment]()

**Comment by Mehrabbruno** [comment]
### tasks - `2023.05.26`

- coded app_cover_v0.0.1
   - code the app_cover component - `1h11min`
   - added background pattern - `4min`
   - adding fonts - `39min`
   - create worklog comment, task, issues and pull request - `9min`
   - `@output` 🏭 [app_cover_v0.0.1](https://github.com/dat-ecosystem/dat-ecosystem/pull/8/commits/21ab2bfb50ecb2ed4a4f98d89e7aa1e47c7b93f0)

---

## worklog 

---

## feedback

- coded app_cover_v0.0.1
- added fonts

---

## proposal

**Comment by Mehrabbruno** [comment]
### tasks - `2023.06.03`

- update - app_cover_v0.0.1 component - `42min`
   -add - character and cover images
   - update - theme color
   - code - function to close on window_bar notify
   - create - worklog comment, and task- `11min`
   - `@output` 📦 [`app_cover_v0.0.2`](https://github.com/dat-ecosystem/dat-ecosystem/pull/8/files#diff-f49b6278570268a9809314bb24f99b9805fbf373a827091f9ced8dde4e6fd31d)

---

## worklog 

[worklog-149](https://youtu.be/l0yaPv_6oLI)

---

## feedback

- moved to fonts to parent file

---

## proposal

**Comment by serapath** [comment]
**feedback `2023.06.07`**

The `app_cover_v0.0.1` and `app_cover_v0.0.2` link to the same commit/diff, so not sure what the difference is, but code looks good.

Some changes would be helpful
```js
// ...
shadow.innerHTML = `
    <div class="cover_wrapper">
        <div class="cover_image"></div>
        <div class="content_wrapper">
            WELCOME TO DAT ECOSYSTEM
        </div>
    </div>
    <style> ${get_theme} </style>
`

// Listening to toggle event 
const listen = (props) =>{
    const {active_state} = props
    ;(active_state==='active')?el.style.display = 'none':''
}

// Added background banner cover
const cover_image = shadow.querySelector('.cover_image')
banner_img = document.createElement('img')
banner_img.src = banner_cover
cover_image.append(banner_img)
// ...
```
could be refactored to
```js
// ...
shadow.innerHTML = `
    <div class="cover_wrapper">
        <div class="cover_image">
          <img src="${banner_cover}">
        </div>
        <div class="content_wrapper">
            WELCOME TO DAT ECOSYSTEM
        </div>
    </div>
    <style> ${get_theme} </style>
`

// Listening to toggle event 
const listen = message => {
  const { type, data: props } = message
  const {active_state} = props
  ;(active_state==='active')?el.style.display = 'none':''
}
```

The main importan thing though is, that all messages should have the following format:
`message = { head, refs, type, data, meta }`, where `type` and `data` are here the main important ones.
I think we should probably make a helper module at some point to ensure consistency.

lastly:
```js
    const view_more_btn = sm_text_button({text:'View more (20)'})
    const tell_me_more_btn = sm_text_button({text:'TELL ME MORE'})
    const cover_window = window_bar({
        name:'Learn_about_us.pdf', 
        src: icon_pdf_reader,
        action_buttons: [tell_me_more_btn, view_more_btn]
    }, listen)
```

This is problematic.
Basically one rule that components should always follow is to only pass as parameters what you can `JSON.stringify` (apart from the protocol listener).

So, if instead of the `tell me more` and `view more` buttons, we could pass the 
`['View more (20)', 'TELL ME MORE']` text, that would be better.

and then just construct the `sm_text_button` inside the window_bar component.

Also, if there are multiple types of buttons, then you could instead pass

`[{ type: 'sm_text_button', data: 'View more (20)' }, { type: 'sm_text_button', data: 'TELL ME MORE' }]`

...but again, if all the window bar buttons are always sm_text_button, then just passing an array of text snippets is probably better

**Comment by Mehrabbruno** [comment]
### tasks - `2023.06.23`

- update - app_cover_v0.0.2
   - code protocol and listen - `8min`
   - read feedback - `4min`
   - commit all changed - `13min`
   - create worklog comment, task, issues and pull request - `11min`
   - `@output` 📦 [`app_cover_v0.0.3`](https://github.com/dat-ecosystem/dat-ecosystem/commit/93789028e25f7267f7349fde463ede481b2c8abb#diff-f49b6278570268a9809314bb24f99b9805fbf373a827091f9ced8dde4e6fd31d)

---

## worklog 

[worklog-158](https://youtu.be/hqYz3begQ9k)

---

## feedback

---

## proposal

**Comment by Mehrabbruno** [comment]
### tasks - `2023.12.06`

- update - app_cover_v0.0.3 component - `20min`
   - add `tell me more info`
   - add toggle func for `tell me more` 
   - create and update tasks - `11min`
   - `@output` 📦[`app_cover_v0.0.4`](https://github.com/Mehrabbruno/dat-ecosystem/commit/0646bc875496238e52977391d7942a8a1c4a2cd2#diff-ce6a1412b787b1586cb615c7a365ce7c97af42c8e130cea6342a8aa71d555bb8)

---

## worklog 

[worklog-185](https://youtu.be/3R4yjPM3zqg)

---

## feedback
  
---

## proposal

**Comment by ninabreznik** [comment]
Change the text in the Tell me more section to:

Community for the next generation Web

Dat ecosystem is driven forward by many projects, most self funded. Some of the projects contribute maintainance and development to core pieces of the Dat ecosystem while others create high level applications built on top of p2p protocols.

**Comment by Mehrabbruno** [comment]
### tasks - `2023.12.20`

- update - app_cover_v0.0.4 component
   - updated content - `5min`
   - create and update tasks - `11min`
   - `@output` 📦 [app_cover_v0.0.5](https://github.com/Mehrabbruno/dat-ecosystem/commit/dcde5b75a1678864dee5081cddeea72f911246e3#diff-ce6a1412b787b1586cb615c7a365ce7c97af42c8e130cea6342a8aa71d555bb8)

---

## worklog 

[worklog-188](https://youtu.be/K7D_jrcDjIc)

---

## feedback
  
---

## proposal

**Comment by ninabreznik** [comment]
Hi, I checked the changes on my local version and it looks very nice. Awesome work on sorting and other changes. 

Now to the feedback and to answer your questions:
- changing the robot on the footer to green when in dark mode would be great, so they are all following the same design pattern
- projects page looks good to me, the data visualisation loads for me, so I don't get the error you are getting.Do you know why this is happening for you? You don't have a slow internet or weak computer? I wonder if this visualisation creator should do some fix or what is the reason for this on your side (because there might be other visitors having the same issue).
`Invalid status for framebuffer for GPULayer "current_position": 0`
- learn_about_us.pdf has a link to the dat-ecosystem visualisation - it should link to Projects section

**Comment by ninabreznik** [comment]
I have a bit more about the Timeline section, so I am adding a new comment for it:
- when there is nothing found for the query let's change 'Nothing is selected' to 'No match found'
![Screenshot_2023-12-20_21-38-39](https://github.com/dat-ecosystem/dat-ecosystem/assets/1076427/a86c4fec-ab33-4361-ba32-9a2fdfe72bd5)
- also filtering seems a bit broken sometimes, not sure when, but i.e. if I filter out only `projects` and then search `dat-` and then leave `projects` filter on but delete search term, I get all the projects including dat-data website (which is not a project)
![Screenshot_2023-12-20_21-49-44](https://github.com/dat-ecosystem/dat-ecosystem/assets/1076427/537d63bc-601c-491f-bfa0-a8c3f057fe70)

- dat-data webpage card seems to always be selected (pink outline)
![Screenshot_2023-12-20_21-37-49](https://github.com/dat-ecosystem/dat-ecosystem/assets/1076427/96326403-6965-436a-9e05-0c35c0ed1c50)
- also when you filter & search by the time, there is something weird happening
![Screenshot_2023-12-20_21-52-26](https://github.com/dat-ecosystem/dat-ecosystem/assets/1076427/45275e52-5044-4c0d-be4d-9d4a37b3c642)
![Screenshot_2023-12-20_22-38-56](https://github.com/dat-ecosystem/dat-ecosystem/assets/1076427/186f2b22-f3a2-4478-81cb-34b33de001e9)

**Comment by ninabreznik** [comment]
And one more thing. Can we please open Manifesto and Important documents on Info page (when you click info button AND when you click on Important documents from the About us in the Home section.
